rfcs/0148-taskcluster-worker-location.md

```diff
@@ -0,0 +1,48 @@
+# RFC 148 - TASKCLUSTER_WORKER_LOCATION
+* Comments: [#148](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/148)
+* Proposed by: @djmitche
+
+# Summary
+
+This proposal defines an environment variable, `TASKCLUSTER_WORKER_LOCATION`, containing this information in an easily-digested format.
+
+## Motivation
+
+Workers often need information about where they are running, for example to select nearby mirrors or other resources.
+
+Taskcluster does not currently provide this in a documented, supported fashion.
+
+# Details
+
+This proposes to introduce `$TASKCLUSTER_WORKER_LOCATION` as an environment variable available within tasks run by docker-worker and generic-worker.
+
+The content has the format `<cloud>/<details>`.
+The `<cloud>` portion will correspond to the provider types used within worker-manager (e.g., `google`, `amazon`).
+The `<details>` portion is specific to the cloud.
+Tasks not executed in any cloud (e.g., static or standalone workers) can define arbitrary locations.
+Examples:
+* `amazon/us-east-1b`
+* `google/us-east1-c`
+* `onprem/pdx1`
+
+If no location is available, workers will not set this variable.
+
+## Implementation Details
+
+Taskcluster-worker-runner already posesses special-case code for each cloud, and makes queries to those clouds' metadata services.
+As such, it is well-suited to determine the location in each cloud.
+
+It will pass this information on to workers as a configuration option, and the workers will include it in the task environment.
+
```

Comment:
Without having explanations on the "how to parse" I fear this could have unintended affects in future cases...

e.g. if AWS/GCP changes their region naming scheme, etc. Doing regex'y things for this sort of thing could be hard.

I would like an explicit "here is how to tell we're not matched" or some sort of helper library set to get meaningful data out.

This not blocking just a concern.

Lastly can we define a syntax more specifically than "like this" e.g. `<cloud>/<details>` where `cloud` is regex `[a-z]` and `details` are regex `[a-z0-9\-.]` or something... so we know what we are working with when writing parsers.

Reply:
The idea is to "pass through" the information from the cloud provider -- so if GCP decides to start naming its zones differently, that's a breaking change they've made and our users need to deal with it just like any other GCP user.

The RFC uses "like this" and we'd have more detailed documentation in worker-runner, and for stuff that deployments set up by hand, it would be defined by the people building those deployments.

Regarding the regexes -- ok, ok, maybe we should just use JSON :)

I'll make some comments below about a possible JSON format.